services/app-api/src/postgres/contractAndRates/prismaToDomainModel.ts

```diff
@@ -92,19 +135,158 @@ function contractFormDataToDomainModel(
     }
 }
 
-function convertUpdateInfoToDomainModel(
-    info: UpdateInfoTableWithUpdater | null
-): UpdateInfoType | undefined {
-    if (!info) {
-        return undefined
+function draftRatesToDomainModel(
+    draftRates: DraftRateWithRelations[]
+): RateRevision[] {
+    return draftRates.map((dr) => ({
+        id: dr.revisions[0].id,
+        revisionFormData: dr.revisions[0].name,
+    }))
+}
+
+function ratesRevisionsToDomainModel(
+    rateRevisions: RateRevisionTable[]
+): RateRevision[] {
+    return rateRevisions.map((rrev) => ({
+        id: rrev.id,
+        revisionFormData: rrev.name,
+    }))
+}
+
+function draftContractRevToDomainModel(
+    revision: DraftContractRevisionTableWithRelations
+): ContractRevision {
+    return {
+        id: revision.id,
+        createdAt: revision.createdAt,
+        updatedAt: revision.updatedAt,
+        formData: contractFormDataToDomainModel(revision),
+        rateRevisions: draftRatesToDomainModel(revision.draftRates),
     }
+}
+
+function contractRevToDomainModel(
+    revisions: ContractRevisionSet[]
+): ContractRevision[] {
+    const contractRevisions = revisions.map((entry) => ({
+        id: entry.contractRev.id,
+        submitInfo: convertUpdateInfoToDomainModel(entry.submitInfo),
+        unlockInfo: entry.unlockInfo
+            ? convertUpdateInfoToDomainModel(entry.unlockInfo)
+            : undefined,
+        createdAt: entry.contractRev.createdAt,
+        updatedAt: entry.contractRev.updatedAt,
+        formData: contractFormDataToDomainModel(entry.contractRev),
+        rateRevisions: ratesRevisionsToDomainModel(entry.rateRevisions),
+    }))
+
+    return contractRevisions
+}
+
+function draftContractToDomainModel(
+    contract: DraftContractTableWithRelations
+): Contract {
+    const revisions = contract.revisions.map((cr) =>
+        draftContractRevToDomainModel(cr)
+    )
 
     return {
-        updatedAt: info.updatedAt,
-        updatedBy: info.updatedBy.email,
-        updatedReason: info.updatedReason,
+        id: contract.id,
+        status: getContractStatus(contract.revisions),
+        stateCode: contract.stateCode,
+        stateNumber: contract.stateNumber,
+        revisions,
+    }
+}
+
```

Comment:
Can you add a comment here and on the rate one? Suggested text:

contractWithHistoryToDomainModel constructs a history for this particular contract including changes to all of its revisions and all related rate revisions, including added and removed rates